Code/Editor/EditorViewportWidget.cpp

```diff
@@ -2562,130 +2318,123 @@ void EditorViewportWidget::SetFOV(float fov)
-        return AZ::DegToRad(fov);
+        return;
     }
 
-    return m_camFOV;
-}
+    // if they've picked the same camera, then that means they want to toggle
+    if (viewEntityId.IsValid())
+    {
+        // Any such events for game entities should be filtered out by the check above
+        AZ_Error(
+            "EditorViewportWidget",
+            Camera::EditorCameraViewRequestBus::FindFirstHandler(viewEntityId) != nullptr,
+            "Internal logic error - active view changed to an entity which is not an editor camera. "
+            "Please report this as a bug."
+        );
 
-//////////////////////////////////////////////////////////////////////////
-bool EditorViewportWidget::CreateRenderContext()
-{
-    return true;
-}
+        m_viewEntityId = viewEntityId;
+        m_viewSourceType = ViewSourceType::CameraComponent;
+        AZStd::string entityName;
+        AZ::ComponentApplicationBus::BroadcastResult(entityName, &AZ::ComponentApplicationRequests::GetEntityName, viewEntityId);
+        SetName(QString("Camera entity: %1").arg(entityName.c_str()));
 
-//////////////////////////////////////////////////////////////////////////
-void EditorViewportWidget::DestroyRenderContext()
-{
+        PostCameraSet();
+    }
+    else
+    {
+        SetDefaultCamera();
+    }
 }
 
 //////////////////////////////////////////////////////////////////////////
 void EditorViewportWidget::SetDefaultCamera()
 {
-    if (IsDefaultCamera())
-    {
-        return;
-    }
-    ResetToViewSourceType(ViewSourceType::None);
-    GetViewManager()->SetCameraObjectId(m_cameraObjectId);
+    m_viewEntityId.SetInvalid();
+    m_viewSourceType = ViewSourceType::None;
+    GetViewManager()->SetCameraObjectId(GUID_NULL);
     SetName(m_defaultViewName);
     SetViewTM(m_defaultViewTM);
-    PostCameraSet();
-}
 
-//////////////////////////////////////////////////////////////////////////
-bool EditorViewportWidget::IsDefaultCamera() const
-{
-    return m_viewSourceType == ViewSourceType::None;
+    // Synchronize the configured editor viewport FOV to the default camera
+    if (m_viewPane)
+    {
+        const float fov = gSettings.viewports.fDefaultFov;
+        m_viewPane->OnFOVChanged(fov);
+        SetFOV(fov);
+    }
+
+    // Push the default view as the active view
+    auto atomViewportRequests = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
+    if (atomViewportRequests)
+    {
+        const AZ::Name contextName = atomViewportRequests->GetDefaultViewportContextName();
+        atomViewportRequests->PushView(contextName, m_defaultView);
+    }
+
+    PostCameraSet();
 }
 
 //////////////////////////////////////////////////////////////////////////
-void EditorViewportWidget::SetSequenceCamera()
+AZ::RPI::ViewPtr EditorViewportWidget::GetCurrentAtomView() const
 {
-    if (m_viewSourceType == ViewSourceType::SequenceCamera)
+    auto atomViewportRequests = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
```

Comment:
This is incorrect for a non-default viewport context and can be simplified

```suggestion
if (m_renderViewport && m_renderViewport->GetViewportContext())
{
    return m_viewportContext->GetDefaultView();
}
```